docs/TENETS.md

```diff
@@ -0,0 +1,60 @@
+# Charter
+
+> hyper is a protective and efficient HTTP library for all.
+
+# Tenets
+
+Tenets are guiding principles. They guide how decisions are made for the whole project. Ideally, we do all of them all the time. In some cases, though, we may be forced to decide between slightly penalizing one goal or another. In that case, we tend to support those goals that come earlier in the list over those that come later (but every case is different).
+
+## 0. Open
+
+hyper is open source, always. The success of hyper depends on the health of the community building and using it. All contributions are in the open. We don't maintain private versions, and don't include features that aren't useful to others.
+
+[We prioritize kindness][CONDUCT], compassion and empathy towards all contributors. Technical skill is not a substitute for human decency.
+
+[CONDUCT]: https://github.com/hyperium/hyper/blob/master/docs/CODE_OF_CONDUCT.md
+
+### Examples
+
+It's not usually hard for an open source library to stay open and also meet its other priorities. Here's some instances where being **Open** would be more important than **Correct** or **Fast**:
+
+- Say an individual were to bring forward a contribution that makes hyper more correct, or faster, perhaps fixing some serious bug. But in doing so, they also insulted people, harrassed other contributors or users, or shamed everyone for the previous code. They felt their contribution was "invaluable". We would not accept such a contribution, instead banning the user and rewriting the code amongst the kind collaborators of the project.
```

Comment:
First of all, :+1: :+1: :+1: for this example, thank you.

Another example of prioritizing Open over Fast, in a different direction:

- Say someone brings a contribution that adds a new feature useful for performance or correctness, but their work accomplishes this by integrating hyper with a proprietary library. We would not accept such a contribution, because we don't want such a feature limited only to those users willing to compromise openness, and we don't want to bifurcate the ecosystem between those who make that compromise and those who don't.

Reply:
Good example. I originally did have a bullet point here to expand a situation along these lines, but never expanded it. Seems worth doing so <3